### PR TITLE
[DBM for MongoDB] Update DBM for MongoDB beta agent version & new data collected

### PR DESCRIPTION
**Pull request by lu-zhengda**
### What does this PR do? What is the motivation?
This PR updates DBM for MongoDB beta agent to `7.58.0-dbm-mongo-1.6`. With the new beta agent version, we also introduce new MongoDB schemas and indexes analysis feature. 

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: layouts/shortcodes/dbm-mongodb-agent-data-collected.md [hunk 1]
```diff
@@ -16,3 +16,8 @@ Database Monitoring for MongoDB gathers operation samples using the `currentOp`
 ### Replication State Changes
 
 Database Monitoring for MongoDB generates an event each time there is a change in the replication state within the MongoDB instance. This ensures that any changes in replication are promptly detected and reported.
+
+### Collection Schemas and Indexes
+
+Database Monitoring for MongoDB collects inferred schema and indexes of MongoDB collections. This information is used to provide insights into the structure and organization of your collections.
```
Comment (buraizu) [inline layouts/shortcodes/dbm-mongodb-agent-data-collected.md:22]:
```suggestion
Database Monitoring for MongoDB collects inferred schemas and indexes of MongoDB collections. This information is used to provide insights into the structure and organization of your collections.
```

File: layouts/shortcodes/dbm-mongodb-agent-data-collected.md [hunk 2]
```diff
@@ -16,3 +16,8 @@ Database Monitoring for MongoDB gathers operation samples using the `currentOp`
...
+When analyzing MongoDB collections, we collect inferred schema information by sampling documents using the `$sample` aggregation stage. From this analysis, we only gather and send metadata about the schema to Datadog, including field names, field prevalence (how often each field appears), and their respective data types. We do not collect or transmit the actual content of documents or any customer business data. This ensures that sensitive data remains protected while still providing valuable insights into the structure and organization of your collections.
```
Comment (buraizu) [inline layouts/shortcodes/dbm-mongodb-agent-data-collected.md:23]:
```suggestion
When analyzing MongoDB collections, Datadog collects inferred schema information by sampling documents using the `$sample` aggregation stage. From this analysis, only metadata about the schema is gathered and sent to Datadog, including field names, field prevalence (how often each field appears), and their respective data types. Datadog does not collect or transmit the actual content of documents or any customer business data. This ensures that sensitive data remains protected while still providing valuable insights into the structure and organization of your collections.
```

File: layouts/shortcodes/dbm-mongodb-beta-new-features [hunk 3]
```diff
@@ -9,3 +9,8 @@ Database Monitoring for MongoDB captures slow operations from either MongoDB slo
 ### Explain plans
 
 Database Monitoring for MongoDB collects explain plans for current running read operations and slow operations. Explain plans provide detailed insights into the query execution plans, helping you optimize query performance.
+
+### MongoDB Collection Schema and Index Analysis
+
+Database Monitoring for MongoDB now offers schema and index analysis for your MongoDB collections. This feature provides insights into the structure and organization of your data by collecting inferred schema details and index information.
+Using a document sampling approach with the `$sample` aggregation stage, we analyze your collection to identify common fields, their data types, and how often each field appears across the sampled documents (field prevalence). In addition to schema details, Database Monitoring for MongoDB also gathers information on the indexes present, giving you a comprehensive view of how data is structured and optimized for querying.
```
Comment (buraizu) [inline layouts/shortcodes/dbm-mongodb-beta-new-features:16]:
```suggestion
Using a document sampling approach with the `$sample` aggregation stage, Datadog analyzes your collection to identify common fields, their data types, and how often each field appears across the sampled documents (field prevalence). In addition to schema details, Database Monitoring for MongoDB also gathers information on the indexes present, giving you a comprehensive view of how data is structured and optimized for querying.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, I requested a few minor updates per the [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#first-and-second-person-pronouns)

**Review by buraizu — APPROVED** [review]
Thanks, LGTM and approved. Let me know here or in the documentation Slack channel when it's ready to be merged.

File: layouts/shortcodes/dbm-mongodb-agent-data-collected.md [hunk 4]
```diff
@@ -16,3 +16,8 @@ Database Monitoring for MongoDB gathers operation samples using the `currentOp`
 ### Replication State Changes
 
 Database Monitoring for MongoDB generates an event each time there is a change in the replication state within the MongoDB instance. This ensures that any changes in replication are promptly detected and reported.
+
+### Collection Schemas and Indexes
+
+Database Monitoring for MongoDB collects inferred schemas and indexes of MongoDB collections. This information is used to provide insights into the structure and organization of your collections.
+When analyzing MongoDB collections, Datadog collects inferred schema information by sampling documents using the `$sample` aggregation stage. From this analysis, only metadata about the schema is gathered and sent to Datadog, including field names, field prevalence (how often each field appears), and their respective data types. Datadog does not collect or transmit the actual content of documents or any customer business data. This ensures that sensitive data remains protected while still providing valuable insights into the structure and organization of your collections.
```
Comment (caseycull) [inline layouts/shortcodes/dbm-mongodb-agent-data-collected.md:23]: LGTM, I'll ask Noah to take a look as well but good to merge in the meantime

**Review by caseycull — APPROVED** [review]

**Review by bgdeutsch — APPROVED** [review]

**Pull request merged**